AsyncMongoMockClient dosen't suport AsyncIOMotorGridFSBucket

This simple test:
```

import pytest
from mongomock_motor import AsyncMongoMockClient
from motor.motor_asyncio import AsyncIOMotorGridFSBucket

@pytest.mark.asyncio
async def test_dummy():
    mongo_client = AsyncMongoMockClient()
    mongo_db = mongo_client["test_database"]
    grid_fs = AsyncIOMotorGridFSBucket(mongo_db)
```

Provide this error:
self = <motor.motor_asyncio.AsyncIOMotorGridFSBucket object at 0x107e0d710>
database = <motor.motor_asyncio.AsyncIOMotorDatabase object at 0x107b091d0>
bucket_name = 'fs', chunk_size_bytes = 261120, write_concern = None
read_preference = None, collection = None

    def __init__(
        self,
        database,
        bucket_name="fs",
        chunk_size_bytes=DEFAULT_CHUNK_SIZE,
        write_concern=None,
        read_preference=None,
        collection=None,
    ):
        """Create a handle to a GridFS bucket.
    
        Raises :exc:`~pymongo.errors.ConfigurationError` if `write_concern`
        is not acknowledged.
    
        This class conforms to the `GridFS API Spec
        <https://github.com/mongodb/specifications/blob/master/source/gridfs/gridfs-spec.rst>`_
        for MongoDB drivers.
    
        :Parameters:
          - `database`: database to use.
          - `bucket_name` (optional): The name of the bucket. Defaults to 'fs'.
          - `chunk_size_bytes` (optional): The chunk size in bytes. Defaults
            to 255KB.
          - `write_concern` (optional): The
            :class:`~pymongo.write_concern.WriteConcern` to use. If ``None``
            (the default) db.write_concern is used.
          - `read_preference` (optional): The read preference to use. If
            ``None`` (the default) db.read_preference is used.
          - `collection` (optional): Deprecated, an alias for `bucket_name`
            that exists solely to provide backwards compatibility.
    
        .. versionchanged:: 3.0
           Removed support for the `disable_md5` parameter (to match the
           GridFSBucket class in PyMongo).
        .. versionchanged:: 2.1
           Added support for the `bucket_name`, `chunk_size_bytes`,
           `write_concern`, and `read_preference` parameters.
           Deprecated the `collection` parameter which is now an alias to
           `bucket_name` (to match the GridFSBucket class in PyMongo).
        .. versionadded:: 1.0
    
        .. mongodoc:: gridfs
        """
        # Preserve backwards compatibility of "collection" parameter
        if collection is not None:
            warnings.warn(
                'the "collection" parameter is deprecated, use "bucket_name" instead',
                DeprecationWarning,
                stacklevel=2,
            )
            bucket_name = collection
    
        db_class = create_class_with_framework(AgnosticDatabase, self._framework, self.__module__)
    
        if not isinstance(database, db_class):
            raise TypeError(
                "First argument to %s must be  MotorDatabase, not %r" % (self.__class__, database)
            )
    
>       self.io_loop = database.get_io_loop()
E       TypeError: 'AsyncIOMotorCollection' object is not callable

--- a/mongomock_motor/__init__.py
+++ b/mongomock_motor/__init__.py
@@ -27,17 +27,21 @@ def masquerade_class(name):
             @property
             def __class__(self):
                 return target
+
         return Wrapper
+
     return decorator
 
 
 def with_async_methods(source, async_methods):
     def decorator(cls):
         for method_name in async_methods:
+
             def make_wrapper(method_name):
                 async def wrapper(self, *args, **kwargs):
                     proxy_source = self.__dict__.get(f'_{cls.__name__}{source}')
                     return getattr(proxy_source, method_name)(*args, **kwargs)
+
                 return wrapper
 
             setattr(cls, method_name, make_wrapper(method_name))
@@ -50,11 +54,13 @@ def with_async_methods(source, async_methods):
 def with_cursor_chaining_methods(source, chaining_methods):
     def decorator(cls):
         for method_name in chaining_methods:
+
             def make_wrapper(method_name):
                 def wrapper(self, *args, **kwargs):
                     proxy_source = self.__dict__.get(f'_{cls.__name__}{source}')
                     getattr(proxy_source, method_name)(*args, **kwargs)
                     return self
+
                 return wrapper
 
             setattr(cls, method_name, make_wrapper(method_name))
@@ -65,24 +71,27 @@ def with_cursor_chaining_methods(source, chaining_methods):
 
 
 @masquerade_class('motor.motor_asyncio.AsyncIOMotorCursor')
-@with_cursor_chaining_methods('__cursor', [
-    'add_option',
-    'allow_disk_use',
-    'collation',
-    'comment',
-    'hint',
-    'limit',
-    'max_await_time_ms',
-    'max_scan',
-    'max_time_ms',
-    'max',
-    'min',
-    'remove_option',
-    'skip',
-    'sort',
-    'where',
-])
-class AsyncCursor():
+@with_cursor_chaining_methods(
+    '__cursor',
+    [
+        'add_option',
+        'allow_disk_use',
+        'collation',
+        'comment',
+        'hint',
+        'limit',
+        'max_await_time_ms',
+        'max_scan',
+        'max_time_ms',
+        'max',
+        'min',
+        'remove_option',
+        'skip',
+        'sort',
+        'where',
+    ],
+)
+class AsyncCursor:
     def __init__(self, cursor):
         self.__cursor = cursor
 
@@ -111,7 +120,7 @@ class AsyncCursor():
 
 
 @masquerade_class('motor.motor_asyncio.AsyncIOMotorLatentCommandCursor')
-class AsyncLatentCommandCursor():
+class AsyncLatentCommandCursor:
     def __init__(self, cursor):
         self.__cursor = cursor
 
@@ -134,39 +143,42 @@ class AsyncLatentCommandCursor():
 
 
 @masquerade_class('motor.motor_asyncio.AsyncIOMotorCollection')
-@with_async_methods('__collection', [
-    'bulk_write',
-    'count_documents',
-    'count',  # deprecated
-    'create_index',
-    'create_indexes',
-    'delete_many',
-    'delete_one',
-    'drop_index',
-    'drop_indexes',
-    'drop',
-    'distinct',
-    'ensure_index',
-    'estimated_document_count',
-    'find_and_modify',  # deprecated
-    'find_one_and_delete',
-    'find_one_and_replace',
-    'find_one_and_update',
-    'find_one',
-    'index_information',
-    'inline_map_reduce',
-    'insert_many',
-    'insert_one',
-    'map_reduce',
-    'options',
-    'reindex',
-    'rename',
-    'replace_one',
-    'save',
-    'update_many',
-    'update_one',
-])
-class AsyncMongoMockCollection():
+@with_async_methods(
+    '__collection',
+    [
+        'bulk_write',
+        'count_documents',
+        'count',  # deprecated
+        'create_index',
+        'create_indexes',
+        'delete_many',
+        'delete_one',
+        'drop_index',
+        'drop_indexes',
+        'drop',
+        'distinct',
+        'ensure_index',
+        'estimated_document_count',
+        'find_and_modify',  # deprecated
+        'find_one_and_delete',
+        'find_one_and_replace',
+        'find_one_and_update',
+        'find_one',
+        'index_information',
+        'inline_map_reduce',
+        'insert_many',
+        'insert_one',
+        'map_reduce',
+        'options',
+        'reindex',
+        'rename',
+        'replace_one',
+        'save',
+        'update_many',
+        'update_one',
+    ],
+)
+class AsyncMongoMockCollection:
     def __init__(self, database, collection):
         self.database = database
         self.__collection = collection
@@ -191,14 +203,17 @@ class AsyncMongoMockCollection():
 
 
 @masquerade_class('motor.motor_asyncio.AsyncIOMotorDatabase')
-@with_async_methods('__database', [
-    'create_collection',
-    'dereference',
-    'drop_collection',
-    'list_collection_names',
-    'validate_collection',
-])
-class AsyncMongoMockDatabase():
+@with_async_methods(
+    '__database',
+    [
+        'create_collection',
+        'dereference',
+        'drop_collection',
+        'list_collection_names',
+        'validate_collection',
+    ],
+)
+class AsyncMongoMockDatabase:
     def __init__(self, client, database, mock_build_info=None):
         self.client = client
         self.__database = database
@@ -234,7 +249,11 @@ class AsyncMongoMockDatabase():
                 raise
             if isinstance(args[0], str) and args[0].lower() == 'buildinfo':
                 return self.__build_info
-            if isinstance(args[0], dict) and args[0] and list(args[0])[0].lower() == 'buildinfo':
+            if (
+                isinstance(args[0], dict)
+                and args[0]
+                and list(args[0])[0].lower() == 'buildinfo'
+            ):
                 return self.__build_info
             raise
 
@@ -252,15 +271,27 @@ class AsyncMongoMockDatabase():
 
 
 @masquerade_class('motor.motor_asyncio.AsyncIOMotorClient')
-@with_async_methods('__client', [
-    'drop_database',
-    'list_database_names',
-    'list_databases',
-    'server_info',
-])
-class AsyncMongoMockClient():
-    def __init__(self, *args, mock_build_info=None, mock_mongo_client=None, mock_io_loop=None, **kwargs):
-        self.__client = _patch_client_internals(mock_mongo_client or MongoClient(*args, **kwargs))
+@with_async_methods(
+    '__client',
+    [
+        'drop_database',
+        'list_database_names',
+        'list_databases',
+        'server_info',
+    ],
+)
+class AsyncMongoMockClient:
+    def __init__(
+        self,
+        *args,
+        mock_build_info=None,
+        mock_mongo_client=None,
+        mock_io_loop=None,
+        **kwargs,
+    ):
+        self.__client = _patch_client_internals(
+            mock_mongo_client or MongoClient(*args, **kwargs)
+        )
         self.__build_info = mock_build_info
         self.__io_loop = mock_io_loop
 
@@ -290,7 +321,11 @@ class AsyncMongoMockClient():
 @contextmanager
 def enabled_gridfs_integration():
     with ExitStack() as stack:
-        stack.enter_context(patch('gridfs.Database', (PyMongoDatabase, MongoMockDatabase)))
-        stack.enter_context(patch('gridfs.grid_file.Collection', (PyMongoDatabase, MongoMockCollection)))
+        stack.enter_context(
+            patch('gridfs.Database', (PyMongoDatabase, MongoMockDatabase))
+        )
+        stack.enter_context(
+            patch('gridfs.grid_file.Collection', (PyMongoDatabase, MongoMockCollection))
+        )
         stack.enter_context(patch('gridfs.GridOutCursor', _create_grid_out_cursor))
         yield

--- a/mongomock_motor/__init__.py
+++ b/mongomock_motor/__init__.py
@@ -1,9 +1,16 @@
+import asyncio
 import importlib
+from contextlib import ExitStack, contextmanager
 from functools import wraps
+from unittest.mock import patch
 
+from mongomock import Collection as MongoMockCollection
+from mongomock import Database as MongoMockDatabase
 from mongomock import MongoClient
+from mongomock.gridfs import _create_grid_out_cursor
+from pymongo.database import Database as PyMongoDatabase
 
-from .patches import _patch_collection_internals
+from .patches import _patch_client_internals, _patch_collection_internals
 
 
 def masquerade_class(name):
@@ -164,6 +171,9 @@ class AsyncMongoMockCollection():
         self.database = database
         self.__collection = collection
 
+    def get_io_loop(self):
+        return self.database.get_io_loop()
+
     def __eq__(self, other):
         return self.__collection == other.__collection
 
@@ -197,6 +207,13 @@ class AsyncMongoMockDatabase():
             'version': '5.0.5',
             'versionArray': [5, 0, 5],
         }
+
+    @property
+    def delegate(self):
+        return self.__database
+
+    def get_io_loop(self):
+        return self.client.get_io_loop()
 
     def get_collection(self, *args, **kwargs):
         return AsyncMongoMockCollection(
@@ -242,9 +259,13 @@ class AsyncMongoMockDatabase():
     'server_info',
 ])
 class AsyncMongoMockClient():
-    def __init__(self, *args, mock_build_info=None, mock_mongo_client=None, **kwargs):
-        self.__client = mock_mongo_client or MongoClient(*args, **kwargs)
+    def __init__(self, *args, mock_build_info=None, mock_mongo_client=None, mock_io_loop=None, **kwargs):
+        self.__client = _patch_client_internals(mock_mongo_client or MongoClient(*args, **kwargs))
         self.__build_info = mock_build_info
+        self.__io_loop = mock_io_loop
+
+    def get_io_loop(self):
+        return self.__io_loop or asyncio.get_event_loop()
 
     def get_database(self, *args, **kwargs):
         return AsyncMongoMockDatabase(
@@ -264,3 +285,12 @@ class AsyncMongoMockClient():
             return getattr(self.__client, name)
 
         return self.get_database(name)
+
+
+@contextmanager
+def enabled_gridfs_integration():
+    with ExitStack() as stack:
+        stack.enter_context(patch('gridfs.Database', (PyMongoDatabase, MongoMockDatabase)))
+        stack.enter_context(patch('gridfs.grid_file.Collection', (PyMongoDatabase, MongoMockCollection)))
+        stack.enter_context(patch('gridfs.GridOutCursor', _create_grid_out_cursor))
+        yield

--- a/mongomock_motor/patches.py
+++ b/mongomock_motor/patches.py
@@ -58,10 +58,13 @@ def _patch_insert_and_ensure_uniques(collection):
                 return fn(data, *args, **kwargs)
             except DuplicateKeyError as exc:
                 raise _provide_error_details(collection, data, exc)
+
         return wrapper
 
     collection._insert = with_enriched_duplicate_key_error(collection._insert)
-    collection._ensure_uniques = with_enriched_duplicate_key_error(collection._ensure_uniques)
+    collection._ensure_uniques = with_enriched_duplicate_key_error(
+        collection._ensure_uniques
+    )
 
     return collection
 

--- a/mongomock_motor/patches.py
+++ b/mongomock_motor/patches.py
@@ -1,5 +1,7 @@
-from mongomock import DuplicateKeyError, helpers
 from functools import wraps
+from unittest.mock import Mock
+
+from mongomock import DuplicateKeyError, helpers
 
 try:
     from beanie.odm.fields import ExpressionField
@@ -100,4 +102,9 @@ def _patch_collection_internals(collection):
     return collection
 
 
-__all__ = ['_patch_collection_internals']
+def _patch_client_internals(client):
+    client.options = Mock(timeout=None)
+    return client
+
+
+__all__ = ['_patch_collection_internals', '_patch_client_internals']

--- a/ruff.toml
+++ b/ruff.toml
@@ -1,0 +1,4 @@
+target-version = "py38"
+
+[format]
+quote-style = "single"

--- a/tests/test_gridfs.py
+++ b/tests/test_gridfs.py
@@ -1,0 +1,18 @@
+import pytest
+from mongomock_motor import AsyncMongoMockClient, enabled_gridfs_integration
+from motor.motor_asyncio import AsyncIOMotorGridFSBucket
+import io
+
+
+@pytest.mark.anyio
+async def test_gridfs():
+    with enabled_gridfs_integration():
+        fs = AsyncIOMotorGridFSBucket(AsyncMongoMockClient()['db'])
+
+        file_name = 'file.txt'
+        file_bytes = b'sup'
+        file_id = await fs.upload_from_stream(file_name, file_bytes)
+
+        with io.BytesIO() as buffer:
+            await fs.download_to_stream(file_id, buffer)
+            assert buffer.getvalue() == file_bytes

--- a/tests/test_gridfs.py
+++ b/tests/test_gridfs.py
@@ -1,7 +1,9 @@
-import pytest
-from mongomock_motor import AsyncMongoMockClient, enabled_gridfs_integration
-from motor.motor_asyncio import AsyncIOMotorGridFSBucket
 import io
+
+import pytest
+from motor.motor_asyncio import AsyncIOMotorGridFSBucket
+
+from mongomock_motor import AsyncMongoMockClient, enabled_gridfs_integration
 
 
 @pytest.mark.anyio

